draft-ietf-mls-protocol.md

```diff
@@ -914,6 +905,58 @@ np[0] -> 1       5[C]
 After performing these operations, the generator of the UpdatePath MUST
 delete the leaf_secret.
 
+## Adding and Removing Leaves
+
+In addition to the path-based updates to the tree described above, it is also
+necessary to add and remove leaves of the tree in order to reflect members being
+added to or removed from the group.  Leaves are always added and removed at the
+right edge of the tree: Either a new rightmost leaf is added, or the rightmost
+leaf is removed.  Nodes' parent/child node relationships are then updated to
+maintain the tree's left-balanced structure.  These operations are also known as
+_extending_ and _truncating_ the tree.
+
+To add a new leaf: Add leaf as the new rightmost leaf of the tree.  Add a blank
+parent node whose right child is the new leaf.  The parent is attached to the
+tree as the right child of the appropriate node to make the updated tree
+left-balanced (or set as a new root).  The former right child of the parent's
+parent becomes the parent's left child (or the old root becomes the left child).
```

Comment:
I suggest giving names to make it easier to read. Also add emphasis that there is a unique way to incorporate the new parent in the tree that keeps it left-balanced (-> " […] as the only appropriate node […]").

```suggestion
To add a new leaf: Add leaf L as the new rightmost leaf of the tree.  Add
a blank parent node P whose right child is L.  P is attached to the
tree as the right child of the only appropriate node to make the updated tree
left-balanced (or set it as a new root).  The former right child of the P's
parent becomes P's left child (or the old root becomes the P's left child if
P is the new root).
```